protocol/validation/tx.go

```diff
@@ -48,6 +48,13 @@ func ConfirmTx(snapshot *state.Snapshot, block *bc.Block, tx *bc.Tx) error {
 			if txin.AssetVersion != 1 {
 				continue
 			}
+			if ii.InitialBlock != initialBlockHash {
+				return errors.WithDetail(ErrBadTx, "issuance is for different blockchain")
+			}
+			assetID := bc.ComputeAssetID(ii.IssuanceProgram, initialBlockHash, ii.VMVersion)
```

Comment:
Hrmm, the exact words from the spec are:
> Compute asset ID from the initial block ID, asset version 1, and the VM version and issuance program declared in the witness. If the resulting asset ID is not equal to the declared asset ID in the issuance commitment, halt and return false.

I think maybe we need to check that here: https://github.com/chain/chain/blob/main/protocol/bc/txinput.go#L181

Currently, we just read the asset ID from the commitment and disregard it.